src/robot/running/arguments/typeconverters.py

```diff
@@ -390,16 +391,42 @@ class ListConverter(TypeConverter):
     abc = Sequence
     value_types = (str, Sequence)
 
-    def no_conversion_needed(self, value):
-        if isinstance(value, str):
+    def __init__(self, type_, custom_converters=None, languages=None):
+        super().__init__(type_, custom_converters, languages)
+        self.nested_types = getattr(type_, '__args__', ())
+
+    @classmethod
+    def handles(cls, type_):
+        base_type = getattr(type_, '__origin__', type_)
+        handled = (cls.type, cls.abc) if cls.abc else cls.type
+        if base_type is tuple:
             return False
-        return super().no_conversion_needed(value)
+        return isinstance(base_type, type) and issubclass(base_type, handled)
+
+    def no_conversion_needed(self, value):
+        return not self.nested_types \
+               and isinstance(value, self.used_type) \
+               and not isinstance(value, str)
 
     def _non_string_convert(self, value, explicit_type=True):
-        return list(value)
+        if not is_list_like(value):
+            raise ValueError
+        return self.__nested_conversion(value, explicit_type)
```

Comment:
If I remember this logic right, this method isn't called if this converter doesn't handle the value and in that case `is_list_like` check would be unnecessary. That would allow moving all code from  `__nested_conversion` here.